src/lib/Microsoft.Health.Events/EventCheckpointing/StorageCheckpointClient.cs

```diff
@@ -161,5 +168,28 @@ public async Task PublishCheckpointAsync(string partitionId)
             Checkpoint checkpoint = _checkpoints[partitionId];
             await UpdateCheckpointAsync(checkpoint);
         }
+
+        /// <summary>
+        /// Deletes the previously recorded checkpoints if the current checkpoint blob path (corresponding to the input event hub) has changed.
+        /// </summary>
+        public async Task ResetCheckpointsAsync()
+        {
+            try
+            {
+                foreach (BlobItem blob in _storageClient.GetBlobs(states: BlobStates.All, prefix: _blobCheckpointPrefix, cancellationToken: CancellationToken.None))
```

Comment:
I'm not sure that this matters, but if the options.BlobPrefix ever changes, previous checkpoints will not be deleted. The only way I can see to address this issue would be to delete all of the blobs in the container, but then we would need 1 container per StorageCheckpointClient.

Reply:
orhpaned prefixes aren't a concern if we as long as the value isn't revert back and they become in use again.

Reply:
Yes, that is correct that if the blob prefix changes, then checkpoints in the previous prefix will not be deleted, however, we will be reading from the new location and so the data ingestion will still be reset.
Having said that, I don't anticipate the prefix changing, right now in PaaS, we seem to be hardcoding the blobprefix with "devicedata", which actually wasn't being honored until now (we were always setting the prefix to app type eg: Normalization). The new prefix now will be *apptype*/*checkpoint prefix* while it was previously *apptype*/
So we will not be deleting the checkpoints that were already created till now (which I believe is okay(?) given we don't have many active iot connectors created through PaaS), but going forward since we don't expect to change the prefix, things should be cleared as intended.